src/lib/matches.js

```diff
@@ -5,5 +5,5 @@ export default function matches(str, pattern, modifiers) {
   if (Object.prototype.toString.call(pattern) !== '[object RegExp]') {
     pattern = new RegExp(pattern, modifiers);
   }
-  return pattern.test(str);
+  return !!str.match(pattern);
```

Comment:
`.test` keeps the state when used along with a regex that has the global or the sticky flag. We can prevent that behaviour getting stateless validation using `.match`

Reply:
What about using `Array.isArray(str.match(pattern))` or `str.match(pattern) !== null` since the possible return values of `.matches` are Array or null?

Reply:
I honestly find that excessively verbose and probably won't be a blocker for the pr. Let's see what the maintainers think.